### ISSUE TITLE
receive error on empty udp messages

### ISSUE THREAD

**Issue by bo-tato** [usocket/usocket#101]
I was doing protohackers challenges and for challenge 4 one of the edge cases they test is sending empty udp messages
my solution: https://gist.github.com/bo-tato/7062c3770cbe7df105b8aa839a244126
you can test (no account or anything needed) at: https://protohackers.com/problem/4
or to simply trigger the error locally send an empty packet:
```
(with-client-socket (socket nil "127.0.0.1" 1234 :protocol :datagram)
  (socket-send socket "" 0))
```
the socket server crashes with:
> receive error: 0
>    [Condition of type SIMPLE-ERROR]

I think the correct behavior would be to call the handler with an empty buffer rather than erroring out of the whole event loop.
In `server.lisp` I changed it from checking `(plusp n)` to checking `(>= n 0)` and then all protohackers tests passed

**Comment by binghe** [comment]
Hi, I think someone ever reported the same issue and the current latest release of usocket has already the code `(<= 0 n)` in `udp-event-loop` (server.lisp) to support zero-length UDP packets. Could you please confirm this with latest release of usocket? (perhaps it's not yet available from Quicklisp, I'm not sure.)

**Comment by bo-tato** [comment]
ah thanks, sorry I hadn't checked git. I was using the version from quicklisp but it's fixed in the latest version as you say.

I also ran into another issue that I've checked now and it's present also in the latest version, but I asked [in a forum](https://www.reddit.com/r/lisp/comments/130o9w8/networking_and_threads/) first as I'm not entirely sure it's a bug in usocket and not my misunderstanding something but haven't got a response yet. Basically it seems calling `close` on a stream and `socket-close` on a socket, do not actually close the connection if another thread is also reading from that stream
